WordPress/openverse
# Replace popper.js with floating-ui

## Fixes



Fixes #506 by @zackkrida 

## Description


This PR replaces the older `popper.js` library with its updated version, `floating-ui`. This is a preparatory step for #2840 and is also useful for updating to Nuxt 3.

## Testing Instructions

The popovers should work as they did before.
The differences are:
RTL placement works correctly. You can see in the homepage snapshots that the popover is aligned with the end of the search type button in both LTR and RTL, whereas in the old popovers, the RTL was aligned with the start of the button.
I removed the `stable` setting for scroll gutter. It was causing the blank space in the inline-end side of the search types popover.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
